### PR TITLE
Fix PE-D bugs and update DG

### PR DESCRIPTION
**Pull request by dez2003**

### CODE REVIEW

**Review by YANYAN-w — APPROVED** [review]
lgtm^_^, can resolve the conflicts then merge

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -591,6 +589,22 @@ Use case ends.
...
     Use case ends.
 
+**Use case: Sort Students**
+
+**MSS**
+
+1. User requests to list students.
+2. T_Assistant shows a list of students.
+3. User requests to sort students.
+4. T_Assistant sorts students in ASCII order.
```
Comment (ghos7ie) [inline docs/DeveloperGuide.md:599]: Sorry I'm blocking this PR, can you update that the message stated in the UG is wrong too

**Review by ghos7ie — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ghos7ie) [reply docs/DeveloperGuide.md:599]: For both sort student and group in the UG, currently the screenshots show that the it is alphabetical order.

Can you add a note that states it is ASCII not alphabetical and that the ss is wrong

Comment (dez2003) [reply docs/DeveloperGuide.md:599]: Ok, should I just update the screenshot in that case?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -822,6 +834,23 @@ Use case ends.
 
       Use case ends.
 
+**Use case: Sort Groups**
+
+**MSS**
+
+1. User lists all groups.
```
Comment (ghos7ie) [inline docs/DeveloperGuide.md:841]: Also same issue, User requests to list all groups

Comment (ghos7ie) [reply docs/DeveloperGuide.md:841]: Please double check the other parts

**Review by ghos7ie — APPROVED** [review]
Not sure if you shifted mark task also, do check on that #415

**Comment by dez2003** [comment]
Closes #415 
Closes #372 
Closes #361
Closes #383

**Pull request merged**